src/latent_gp.jl

```diff
@@ -0,0 +1,32 @@
+"""
+    LatentGP(f<:GP, x<:AbstractVector, Φ)
+
+ - `fx` is a `FiniteGP`.
+ - `Φ` is the log likelihood function which maps input of type `Tx` to `Real`.
+    
+"""
+struct LatentGP{T<:AbstractGPs.FiniteGP}
+    fx::T
+    Φ
+    function LatentGP(fx::AbstractGPs.FiniteGP, Φ) where Tx
+        return new{typeof(fx)}(fx, Φ)
+    end    
+end
+
+function LatentGP(f::GP, x::AbstractVector{Tx}, σ², Φ) where Tx
+        return LatentGP(f(x, σ²), Φ)
+end
+
+function LatentGP(f::GP, x::AbstractVector{Tx}, Φ) where Tx
+        return LatentGP(f(x), Φ)
+end
```

Comment:
Do we really need these additional methods? Is it not sufficient to have
```julia
LatentGP(f(x, σ²), Φ)
```
as the unique + correct way to construct a `LatentGP`?

Reply:
Nope. Now that you point it out, they seem unnecessary.